fix: enable locale search by all locale display options then save locale code

prestonb91 opened the pull request:
✅ Resolves #1053 
- [x] PR title follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specifications
- [x] PR assignee has been selected
- [x] PR label has been selected
- [x] @NabbeunNabi has been selected for preliminary review

## 🔧 What changed
- ModEditHealthcareProfessionalSection only searched by filtering through language codes
- This prevented users from searching with the full locale name text (i.e. English) or to search by the local language (i.e. 日本語 (Japan)
- Updated handleLocaleInputChange to filter across code, simple text, display text based on input value, then map the locale code to filtered items so that it still saves the locale code

## 📸 Screenshots
-   ### Before 
<img width="217" alt="Locale-search-bug-1" src="https://github.com/user-attachments/assets/4b2bade7-94c3-4fa8-a3ce-dd9953d817ae" />
<img width="211" alt="Locale-search-bug-2" src="https://github.com/user-attachments/assets/2c923bf0-d1ce-46a9-8ae1-00bdb16aad39" />

-   ### After
https://github.com/user-attachments/assets/abf7b85b-eb99-4ce7-bd16-0d77812e5412

NabbeunNabi reviewed (changes requested):
This is a good idea.

We do have a store that has functions to deal with locale values.  It has the simple text and code. Could you make a reusable store function with this type of logic where you can input a language and it will use the actually language and then return the code so you can set the value that way and put a comment above it.

Here is the file. https://github.com/ourjapanlife/findadoc-web/blob/main/stores/localeStore.ts

Great start!

prestonb91 commented:
@NabbeunNabi thanks for the feedback. 
I refactored to create a reusable function getLocaleByInput in the localStore and adjusted the ModEditHealthcareProfessionalSection to use it. 
Manually tested and it searches by display options and saves like the original commit. 
https://github.com/ourjapanlife/findadoc-web/pull/1062/commits/34ad046f4a52846fe024bbce7de4db98a9d91d75

Wanted to confirm if this is what you were looking for based on your comments?

NabbeunNabi commented on stores/localeStore.ts:
```diff
@@ -30,7 +30,26 @@ export const useLocaleStore = defineStore('locale', () => {
     }
 
-    return { locale, localeDisplayOptions, mvpLocaleDisplayOptions, setLocale, formatLanguages, formatLanguageCodeToSimpleText }
+    // Takes the input value and searches all locale display options to return the locale code
+    function getLocaleByInput(inputValue: string) {
+        const filteredLocales = localeDisplayOptions
```
Great job! I would think along with the comment you can name the function `getLocaleCodeBySimpleTextInput` or something similar to this for the name to know I am extracting the code by the language name.

prestonb91 replied:
Okay! I updated it to “getLocaleByNameInput"
Just to clarify, didn’t name it get by simpleText because this will also look up by code and displayText. 
Reason was, for example, if it was only simple text, it would not pick up Japan locale unless typed out in kanji, same with cases of korean or hebrew, etc where all simpleText is not in English text. 
So to make the search more robust, it will pick up the locale with its code or simpleText or displayText.

NabbeunNabi reviewed (approved):
@prestonb91 just one thing to change on my end but I'm sure you will finish by the time @ermish reviews it.

So on to him!

prestonb91 commented on stores/localeStore.ts:
```diff
@@ -30,7 +30,26 @@ export const useLocaleStore = defineStore('locale', () => {
+    function getLocaleByNameInput(inputValue: string) {
+        const filteredLocales = localeDisplayOptions
+            .filter(localeDisplay =>
+                localeDisplay.code.toLowerCase().replace('_', '').includes(inputValue)
```
@ermish thank you for the feedback and have just made the changes!

ermish reviewed (approved):
Looks great! Nice work @prestonb91 :shipit:
